tests/h/assets_test.py

```diff
@@ -58,6 +67,22 @@ def test_environment_generates_bundle_urls(mtime):
     ]
 
 
+@patch(open_target, _fake_open)
+@patch('os.path.getmtime')
+def test_environment_version_returns_version_if_asset_exists(mtime):
```

Comment:
There is some boilerplate in the tests for `Environment` that could be removed if the tests were grouped into a class. I will save this for a follow-up PR.